src/components/app.tsx

```diff
@@ -122,29 +136,34 @@ export class App extends React.PureComponent<IProps, IState> {
           activityName={activity.name}
           singlePage={activity.layout === ActivityLayouts.SinglePage}
         />
-        <ActivityNavHeader
-          activityPages={activity.pages}
-          currentPage={currentPage}
-          fullWidth={fullWidth}
-          onPageChange={this.handleChangePage}
-          singlePage={activity.layout === ActivityLayouts.SinglePage}
-        />
-        { activity.layout === ActivityLayouts.SinglePage
-          ? this.renderSinglePageContent(activity)
-          : currentPage === 0
-            ? this.renderIntroductionContent(activity)
-            : activity.pages[currentPage - 1].is_completion
-              ? this.renderCompletionContent(activity)
-              : <ActivityPageContent
-                  enableReportButton={currentPage === activity.pages.length && enableReportButton(activity)}
-                  isFirstActivityPage={currentPage === 1}
-                  isLastActivityPage={currentPage === activity.pages.filter((page) => !page.is_hidden).length}
-                  pageNumber={currentPage}
-                  onPageChange={this.handleChangePage}
-                  page={activity.pages.filter((page) => !page.is_hidden)[currentPage - 1]}
-                  totalPreviousQuestions={totalPreviousQuestions}
-                  teacherEditionMode={this.state.teacherEditionMode}
-                />
+        { authError === ""
+          ? <ActivityNavHeader
+              activityPages={activity.pages}
+              currentPage={currentPage}
+              fullWidth={fullWidth}
+              onPageChange={this.handleChangePage}
+              singlePage={activity.layout === ActivityLayouts.SinglePage}
+            />
+          : ""
```

Comment:
You could simplify this a little bit and remove the empty string in the condition with:
```
        { authError === "" &&
          <ActivityNavHeader
            activityPages={activity.pages}
            currentPage={currentPage}
            fullWidth={fullWidth}
            onPageChange={this.handleChangePage}
            singlePage={activity.layout === ActivityLayouts.SinglePage}
          />
        }
```

Reply:
Or even:
```
        { authError &&
          ...
```